android/library/src/test/java/com/localytics/library/MediaTrackerTest.java

```diff
@@ -84,35 +82,56 @@ public void tagEvent(String eventName, Map<String, String> videoAttributes) {
         mediaTracker.tagEventAtTime(1000000);
     }
 
+
     @Test
-    public void testStopBeforeStartFailure() throws Exception {
+    public void testDoubleOverlap() throws Exception {
         MediaTracker mediaTracker = MediaTracker.create(1000000, new EventTagger() {
             @Override
-            public void tagEvent(String eventName, Map<String, String> videoAttributes) { }
+            public void tagEvent(String eventName, Map<String, String> videoAttributes) {
+                assertEquals("1000", videoAttributes.get(MediaTracker.MEDIA_LENGTH_SECONDS));
+                assertEquals("450", videoAttributes.get(MediaTracker.TIME_PLAYED_SECONDS));
+                assertEquals("45", videoAttributes.get(MediaTracker.PERCENT_PLAYED));
+            }
         });
-        boolean exceptionCaught = false;
-        try {
-            mediaTracker.stopAtTime(500000);
-        } catch (Exception e) {
-            exceptionCaught = true;
-        }
-        assertTrue(exceptionCaught);
+
+        //range 1
+        mediaTracker.playAtTime(50000);
+        mediaTracker.stopAtTime(150000);
+        //range 2
+        mediaTracker.playAtTime(300000);
+        mediaTracker.stopAtTime(500000);
+        //overlaps both ranges
+        mediaTracker.playAtTime(100000);
+        mediaTracker.stopAtTime(350000);
+
+        mediaTracker.tagEventAtTime(500000);
     }
 
     @Test
-    public void testStopAfterMediaLengthFailure() throws Exception {
+    public void testCompleteOverlap() throws Exception {
         MediaTracker mediaTracker = MediaTracker.create(1000000, new EventTagger() {
             @Override
-            public void tagEvent(String eventName, Map<String, String> videoAttributes) { }
+            public void tagEvent(String eventName, Map<String, String> videoAttributes) {
+                assertEquals("1000", videoAttributes.get(MediaTracker.MEDIA_LENGTH_SECONDS));
+                assertEquals("800", videoAttributes.get(MediaTracker.TIME_PLAYED_SECONDS));
+                assertEquals("80", videoAttributes.get(MediaTracker.PERCENT_PLAYED));
+            }
         });
-        boolean exceptionCaught = false;
-        try {
-            mediaTracker.playAtTime(0);
-            mediaTracker.stopAtTime(5000000);
-        } catch (Exception e) {
-            exceptionCaught = true;
-        }
-        assertTrue(exceptionCaught);
+
+        //range 1
+        mediaTracker.playAtTime(50000);
+        mediaTracker.stopAtTime(150000);
+        //range 2
+        mediaTracker.playAtTime(300000);
+        mediaTracker.stopAtTime(500000);
+        //overlaps both ranges
```

Comment:
I'm confused by this comment.   How does `300000-500000` overlap both ranges?   All three ranges are mutually exclusive.